VideoLocker/src/test/java/org/edx/mobile/test/http/ApiTests.java

```diff
@@ -333,6 +322,54 @@ public boolean apply(CourseComponent component) {
                 component = component.getParent();
             }
         }
+
+
     }
 
+    /**
+     * Verifies the parsing of course structure json to {@link CourseComponent} model class.
+     *
+     * @param component Parsed {@link CourseComponent} model.
+     */
+    private void verifyModelParsing(CourseComponent component) {
+        JSONObject jsonObj;
+        try {
+            jsonObj = new JSONObject(MockDataUtil.getMockResponse("get_course_structure"));
+            jsonObj = jsonObj.getJSONObject("blocks");
+            jsonObj = jsonObj.getJSONObject(component.getId());
+            assertNotNull(jsonObj);
+            // Not using the getDisplayName function below, because it returns a placeholder text
+            // when the display_name field's value is empty.
+            assertEquals(jsonObj.getString("display_name"), component.getInternalName());
+            assertEquals(jsonObj.getBoolean("graded"), component.isGraded());
+            assertEquals(jsonObj.getString("student_view_url"), component.getBlockUrl());
+            assertEquals(jsonObj.getBoolean("student_view_multi_device"), component.isMultiDevice());
+            assertEquals(jsonObj.getString("lms_web_url"), component.getWebUrl());
+
+            // Type specific validations
+            Gson gson = new Gson();
+            switch (component.getType()) {
+                case VIDEO: {
+                    String dataRawJson = jsonObj.getJSONObject("student_view_data").toString();
+                    assertTrue(component instanceof VideoBlockModel);
+                    VideoBlockModel model = (VideoBlockModel) component;
+                    VideoData expected = gson.fromJson(dataRawJson, VideoData.class);
+                    assertEquals(expected, model.getData());
+                    break;
+                }
+                case DISCUSSION: {
+                    String dataRawJson = jsonObj.getJSONObject("student_view_data").toString();
+                    assertTrue(component instanceof DiscussionBlockModel);
+                    DiscussionBlockModel model = (DiscussionBlockModel) component;
+                    DiscussionData expected = gson.fromJson(dataRawJson, DiscussionData.class);
+                    assertEquals(expected, model.getData());
+                    break;
+                }
+            }
+        } catch (JSONException e) {
+            fail("Error while converting raw json to JSONObject: " + e.getMessage());
+        } catch (IOException e) {
+            fail("Error reading raw json: " + e.getMessage());
```

Comment:
You could just declare the method to throw these exceptions, since we're not expecting these to happen anyway.